drivers/led/lp50xx.c

```diff
@@ -316,32 +286,77 @@ static int lp50xx_init(const struct device *dev)
 		return err;
 	}
 
-	/* Enable device */
-	err = lp50xx_enable(dev, true);
+	return 0;
+}
+
+static int lp50xx_turn_off(const struct device *dev)
+{
+	int err;
+
+	/* Disable hardware */
+	err = lp50xx_hw_enable(dev, false);
 	if (err < 0) {
-		LOG_ERR("%s: failed to enable", dev->name);
+		LOG_ERR("%s: failed to enable hardware", dev->name);
 		return err;
 	}
 
 	return 0;
 }
 
-#ifdef CONFIG_PM_DEVICE
-static int lp50xx_pm_action(const struct device *dev,
-			    enum pm_device_action action)
+static int lp50xx_pm_action(const struct device *dev, enum pm_device_action action)
 {
 	switch (action) {
 	case PM_DEVICE_ACTION_SUSPEND:
 		return lp50xx_enable(dev, false);
 	case PM_DEVICE_ACTION_RESUME:
 		return lp50xx_enable(dev, true);
+	case PM_DEVICE_ACTION_TURN_ON:
+		return lp50xx_turn_on(dev);
+	case PM_DEVICE_ACTION_TURN_OFF:
+		return lp50xx_turn_off(dev);
```

Comment:
AFAIU the `TURN_OFF` action is called when the power domain is about to turn the power off. So it should be about preparing the device and the driver for the power-off to come (backup the configuration, etc). I don't see the value of powering off the device here. It is happening anyway. What am I missing ?

Reply:
I am not convinced but OK...